src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -366,12 +360,21 @@ private void WriteStreamReferenceProperty(string propertyName, ODataStreamRefere
             ODataProperty property)
         {
             this.WriterValidator.ValidateNullPropertyValue(
-                this.currentPropertyInfo.MetadataType.TypeReference, property.Name, this.Model);
+                this.currentPropertyInfo.MetadataType.TypeReference, property.Name,
+                this.currentPropertyInfo.IsTopLevel, this.Model);
 
             if (this.currentPropertyInfo.IsTopLevel)
             {
-                // TODO: Enable updating top-level properties to null #645
-                throw new ODataException("A null top-level property is not allowed to be serialized.");
+                // From the spec:
+                // 11.2.3 Requesting Individual Properties
+                // ...
+                // If the property is single-valued and has the null value, the service responds with 204 No Content.
+                // ...
+                // ValidateNullPropertyValue will throw by default but might allow this code to be reached
+                // if validation for this scenario is off. In that case, write "value": null which is not
+                // per spec but the best we could do indexer this case.
+                this.JsonWriter.WriteName(JsonLightConstants.ODataValuePropertyName);

```

Comment:
ODL6.x would not write a value property at all in this case. It would include "@odata.null" : true instead.